backend/src/main.py

```
@@ -109,6 +112,30 @@ async def limit_order_loop():
+                if mempool_monitor_service.is_running:
+                    if mempool_monitor_service.monitoring_task is None or mempool_monitor_service.monitoring_task.done():
+                        logger.warning("Mempool monitor task is not running but is_running is True. Restarting...")
+                        await mempool_monitor_service.start_monitoring()
+
+                # Check Auto Trader
+                if auto_trader_service.trading_enabled:
+                    if auto_trader_service.trade_loop_task is None or auto_trader_service.trade_loop_task.done():
+                        logger.warning("Auto trader trade loop is not running but trading_enabled is True. Restarting...")
+                        auto_trader_service.start_trading()
+
+            except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)